modules/aws/main.tf

```diff
@@ -43,6 +43,7 @@ locals {
   s3_kms_key_arn             = length(var.s3_kms_key_arns) > 0 ? var.s3_kms_key_arns : [data.aws_kms_key.s3_default.arn]
   source_identity            = (length(var.source_identities) > 0 ? [{ test : var.source_identity_test, variable : "sts:SourceIdentity", values : var.source_identities }] : [])
   principal_check            = (length(var.streamnative_principal_ids) > 0 ? [{ test : "StringLike", variable : "aws:PrincipalArn", values : var.streamnative_principal_ids }] : [])
+  vendor_federation          = (var.allow_vendor_federation ? [{ test : "StringLike", variable : "aws:FederatedProvider", values : ["accounts.google.com"] }] : [])
```

Comment:
I'll make this change but I think I'm not going to change the var to a list. Adding it as a list and then coalescing that list of conditions doesn't make any sense since you wont be able to have multiple providers for your web identity tokens anyways, and I think that will lead to bugs. We should issue a new version of the module if we change that, since it will be a fundamental access change.

I'm also going to add in a disclaimer that this should _not_ be changed unless explicitly instructed to by StreamNative. This will help us to avoid scenarios where customers update their policy and add in conditions without notifying us as to the change so we can update the supportAccess roleChain.